### PR TITLE
Compute node vnc support

### PR DESCRIPTION
**Pull request by diedpigs**

### CODE REVIEW

File: roles/compute_build_vnfs/tasks/main.yml [hunk 1]
```diff
@@ -54,13 +54,46 @@
...
+       - '@X Window System'
+       - '@Xfce'
 
    - name: copy ssh keys over
      copy: src=cluster_root.pub dest={{ compute_chroot_loc }}/root/.ssh/authorized_keys
 
    - name: copy slurm.conf over
      copy: src=/etc/slurm/slurm.conf dest={{ compute_chroot_loc }}/etc/slurm/slurm.conf
 
+   - name: download TurboVNC rpm
```
[the remark this answers is not in the document]
Comment (diedpigs) [reply roles/compute_build_vnfs/tasks/main.yml]: remember why i did not use repo method. 
```
Downloading packages:
warning: /var/cache/yum/x86_64/7/TurboVNC/packages/turbovnc-2.2.x86_64.rpm: Header V3 DSA/SHA1 Signature, key ID 72feb9ce: NOKEY
Retrieving key from https://pgp.mit.edu/pks/lookup?op=get&search=0x6BBEFA1972FEB9CE


GPG key retrieval failed: [Errno 12] Timeout on http://pgp.mit.edu/pks/lookup?op=get&search=0x6BBEFA1972FEB9CE: (28, 'Operation too slow. Less than 1000 bytes/sec transferred the last 30 seconds')
```

somehow the pgp key verification failed so turbovnc installation failed. they do provide pgp key on their website, we can hard code it in the TurboVNC.repo file, but dont think it's a good idea.

[the remark this answers is not in the document]
Comment (diedpigs) [reply roles/compute_build_vnfs/tasks/main.yml]: Added in 7c9b973

File: roles/compute_build_vnfs/tasks/main.yml [hunk 2]
```diff
@@ -54,13 +54,46 @@
...
+
+   - name: extract Websockify source code into chroot env
+     unarchive:
+       src: /vagrant/CRI_XCBC/files/websockify-0.8.0.tar.gz
+       dest: '{{ compute_chroot_loc }}/tmp'
+
+   - name: install Websockify inside chroot env
+     command: "chroot {{ compute_chroot_loc }} /bin/bash -c 'cd /tmp/websockify-0.8.0; python setup.py install'"
+
+   - name: set compute node to boot with multi-user mode
```
[the remark this answers is not in the document]
Comment (diedpigs) [reply roles/compute_build_vnfs/tasks/main.yml]: Added in e27ab02

**Comment by diedpigs** [comment]
to rerun only this role, you can execute this command:

```
ansible-playbook -c local -i /vagrant/CRI_XCBC/hosts -l `hostname` /vagrant/CRI_XCBC/ohpc.yaml -t compute_build_vnfs -s -v
```

where `-s` is running with sudo, `-v` is running as verbose mode

File: roles/compute_build_vnfs/tasks/main.yml [hunk 3]
```diff
@@ -70,12 +125,14 @@
...
 
+   - name: firewalld on compute image disabled
+     command: chroot '{{ compute_chroot_loc }}' systemctl disable firewalld
+
    - name: chronyd on compute image enabled 
      command: chroot '{{ compute_chroot_loc }}' systemctl enable chronyd
 
    - name: add headnode to compute chrony.conf
      lineinfile: line="server {{ headnode_private_ip }}" dest={{ compute_chroot_loc }}/etc/chrony.conf state=present
-
```
Comment (jprorama) [inline roles/compute_build_vnfs/tasks/main.yml]: Was this line delete intentional?

Comment (diedpigs) [reply roles/compute_build_vnfs/tasks/main.yml]: by mistake

**Pull request merged**